packages/components/src/tooltip/index.js

```diff
@@ -96,7 +96,7 @@ function Tooltip( {
 	children,
 	position,
 	text,
-	shortcut,
+	shortcut = null,
```

Comment:
Why was this change necessary?

Reply:
The TS was complaining that `shortcut` was a required prop, but, looking at all usages of Tooltip, it isn't.

So until we type Tooltip in another PR, I thought this was the least offensive way to make it happy.

Reply:
Got it, thank you for the insight!

I would suggest a slightly different approach which doesn't introduce any runtime changes, while still "tricking" the TS compiler not to display an error — this is achieved by moving the props destructuring inside the component's implementation:

```diff
diff --git a/packages/components/src/tooltip/index.js b/packages/components/src/tooltip/index.js
index deeb1c8c26..85859e1b17 100644
--- a/packages/components/src/tooltip/index.js
+++ b/packages/components/src/tooltip/index.js
@@ -92,13 +92,9 @@ const emitToChild = ( children, eventName, event ) => {
 	}
 };
 
-function Tooltip( {
-	children,
-	position,
-	text,
-	shortcut = null,
-	delay = TOOLTIP_DELAY,
-} ) {
+function Tooltip( props ) {
+	const { children, position, text, shortcut, delay = TOOLTIP_DELAY } = props;
+
 	/**
 	 * Whether a mouse is currently pressed, used in determining whether
 	 * to handle a focus event as displaying the tooltip immediately.
```

Reply:
>The TS was complaining that shortcut was a required prop, but, looking at all usages of Tooltip, it isn't.

Do we know why this was shown? I couldn't find anywhere in GB repo where this prop was deemed required. Same goes to `position` prop I believe.

Reply:
It's not deemed as required, but with the current notation TS _thinks_ it's required, since it doesn't have a default value (and since there isn't any alternative JSDoc to say otherwise).

That's why, if we move the prop destructuring to the function body, TS has to no way to infer the type of `props` and therefore can't throw any errors

Reply:
Thanks for explaining!

Reply:
Thanks for the explanation and the destructuring example 🙇